### PR TITLE
MAINT: Fine-tune versioning extension of sphinx

### PR DESCRIPTION
**Pull request by oesteban**
This PR revises the sphinxcontrib-versioning settings and the CircleCI build for docs:

- [x] Revised package dependencies
  - [x] Remove ``requirements.txt``
  - [x] Check consistency between ``setup.cfg`` and ``docs/requirements.txt``
  - [x] Clean-up some dependencies and remove nilearn (had a couple of calls replaceable by nibabel.load).
- [x] Replaced `long_description.rst` with `README.rst`
- [x] Revised the CircleCI workflow:
  - [x] Deploy docs only on master and tags
  - [x] Build versions only on tags, on master, and on commits which message contains the label `[docs version]`.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -441,18 +440,32 @@ jobs:
           paths:
             - ./docs/_build/_html
       - checkout
+      - run:
+          name: Set up version
+          command: |
+            THISVERSION=$( python get_version.py )
+            sed -i -E "s/'version'/'${CIRCLE_TAG:-$THISVERSION}'/" docs/conf.py
```
Comment (oesteban) [inline .circleci/config.yml]: seems like sphinx-versioning also does this replacement, so I'll probably have to roll back these lines

**Comment by rwblair** [comment]
```
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmp/tmp6lko69sesphinxcontrib_versioning/cf31d7a42ac6ebd2980509d65874ba4af804062c/docs/conf.py", line 24, in <module>
    from smriprep._version import get_versions
ModuleNotFoundError: No module named 'smriprep
```

Something wonky about imports in the git cloned versions doc build. conf.py is modifying syspath but it doesn't seem to work there. In my local version with smriprep installed at they are able to do this import.

**Comment by oesteban** [comment]
Don't worry about this - I still need to push some work I did yesterday. I'll give you a heads up when this is ready for you.

**Comment by oesteban** [comment]
Okay, there it is. The version is resolved on the non-versioned build, but does not resolve well for the versioned one.

**Comment by oesteban** [comment]
Okay, I believe this should work. I'm going to go ahead and merge.

**Pull request merged**